c/public/lib/source/ttread/ttread.c

```diff
@@ -820,21 +993,551 @@ static void hmtxRead(ttrCtx h) {
+
+        srcSeek(h, tupleHeaderOffset);
+        variationDataSize = read2(h);
+        tupleIndex = read2(h);
+
+        if (tupleIndex & gvar_FLAG_EMBEDDED_PEAK_TUPLE) {
+            for (j = 0; j < h->gvar.axisCount; j++)
+                peakTupleCoords[j] = (Fixed)sread2(h) << 2; /* Fixed 2.14 to 16.16 */
+        } else {
+        /* The low 12 bits are an index into a shared tuple records array.
+        so check whether the index is out of range. */
+            uint16_t index = tupleIndex & gvar_FLAG_TUPLE_INDEX_MASK;
+            if (index >= h->gvar.sharedTupleCount)
+                fatal(h, ttrErrBadGlyphData, "tuple count out of range in gvar");
+            if (h->gvar.sharedTuples.cnt > 0)
+                peakTupleCoordsToUse = &h->gvar.sharedTuples.array[index * h->gvar.axisCount];
+            else
+                goto exit; /* Return from here as we don't have peakTupleCoords. */
+        }
+
+        if (tupleIndex & gvar_FLAG_INTERMEDIATE_TUPLE) {
+            /* First read intermediate start coords and then read intermediate end coords for each axis. */
+            for (j = 0; j < h->gvar.axisCount; j++)
+                imStart[j] = (Fixed)sread2(h) << 2; /* convert Fixed 2.14 to 16.16 */
+            for (j = 0; j < h->gvar.axisCount; j++)
+                imEnd[j] = (Fixed)sread2(h) << 2; /* convert Fixed 2.14 to 16.16 */
+        }
+        tupleHeaderOffset = srcTell(h);
+
+        scalar = calculateScalar(h, tupleIndex, peakTupleCoordsToUse, imStart, imEnd);
+
+        if (scalar == 0)
+            continue;
+
+        srcSeek(h, serializedDataOffset);
+        if (tupleIndex & gvar_FLAG_PRIVATE_POINT_NUMBERS) {
+            pntCount = (unsigned long)read1(h);
+
+            /* check whether deltas are applied to all points. */
+            if (pntCount == 0) {
+                bAllPoints = 1;
+                pointIndicesCount = 0;
+            } else {
+                dnaSET_CNT(privatePoints, gvarReadPackedPointNumbers(h, pntCount, &privatePoints.array[0], nPoints));
+                bAllPoints = 0;
+                pointIndices = privatePoints.array;
+                pointIndicesCount = privatePoints.cnt;
+            }
+        } else {
+            if (!bAllSharedPoints) {
+                pointIndices = sharedPoints.array;
+                pointIndicesCount = sharedPoints.cnt;
+            } else
+                pointIndicesCount = 0;
+
+            bAllPoints = bAllSharedPoints;
+        }
+
+        /* read deltas for x and y coordinates. */
+        xIntDeltas.cnt = gvarReadPackedDeltas(h, xIntDeltas.array, pointIndicesCount ? pointIndicesCount : nPoints);
+        if (!xIntDeltas.cnt)
+            continue;
+
+        yIntDeltas.cnt = gvarReadPackedDeltas(h, yIntDeltas.array, pointIndicesCount ? pointIndicesCount : nPoints);
+        if (!yIntDeltas.cnt)
+            continue;
+
+        if (bAllPoints) {
+            bAllPoints = 0;
```

Comment:
The value stored in `bAllPoints` here is never read.

Reply:
correct. removed line